src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3025,6 +3026,7 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
             TypeSymbol parameterType,
             RefKind parRefKind,
             bool ignoreOpenTypes,
+            bool isThisArgument,
```

Comment:
>bool isThisArgument [](start = 12, length = 19)

It  might be better if we simply infer argRefKind for the extension method this argument prior to calling this method instead of making this method special case the scenario. As an additional benefit, we might get better handling around conversions, similar to other parameters, we probably don't want to allow any conversion for **ref this** argument.